src/main/java/ch/njol/skript/effects/EffVisualEffect.java

```diff
@@ -77,50 +80,57 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 		effects = (Expression<VisualEffect>) exprs[base];
 		direction = (Expression<Direction>) exprs[base + 1];
 		where = exprs[base + 2];
-		if (exprs[base + 3] != null) {
-			if (exprs[3].getReturnType() == Player.class)
-				players = (Expression<Player>) exprs[base + 3];
-			else
-				radius = (Expression<Number>) exprs[base + 3];
-		}
+		players = (Expression<Player>) exprs[base + 3];
+		radius = (Expression<Number>) exprs[base + 4];
+
 		if (effects instanceof Literal) {
-			final VisualEffect[] effs = effects.getAll(null);
-			boolean hasLocationEffect = false, hasEntityEffect = false;
-			for (final VisualEffect e : effs) {
-				if (e.isEntityEffect())
+			//noinspection ConstantConditions
+			VisualEffect[] effs = effects.getAll(null);
+
+			boolean hasLocationEffect = false;
+			boolean hasEntityEffect = false;
```

Comment:
those booleans are opposite of each other, I don't think two of them are needed in that case

Reply:
They would be if the effect only allowed a single visual effect, but it supports multiple, therefore the expression can have both a location effect and an entity effect